conans/tools.py

```diff
@@ -137,3 +141,180 @@ def patch(base_path=None, patch_file=None, patch_string=None):
+                        ('wSuiteMask', ctypes.c_ushort),
+                        ('wProductType', ctypes.c_byte),
+                        ('wReserved', ctypes.c_byte)]
+
+        os_version = _OSVERSIONINFOEXW()
+        os_version.dwOSVersionInfoSize = ctypes.sizeof(os_version)
+        retcode = ctypes.windll.Ntdll.RtlGetVersion(ctypes.byref(os_version))
+        if retcode != 0:
+            return None
+
+        return Version("%d.%d" % (os_version.dwMajorVersion, os_version.dwMinorVersion))
+
+    def get_debian_version_name(self, version):
+        if not version:
+            return None
+        elif version.major() == "8.Y.Z":
+            return "jessie"
+        elif version.major() == "7.Y.Z":
+            return "wheezy"
+        elif version.major() == "6.Y.Z":
+            return "squeeze"
+        elif version.major() == "5.Y.Z":
+            return "lenny"
+        elif version.major() == "4.Y.Z":
+            return "etch"
+        elif version.minor() == "3.1.Z":
+            return "sarge"
+        elif version.minor() == "3.0.Z":
+            return "woody"
+
+    def get_win_version_name(self, version):
+        if not version:
+            return None
+        elif version.major() == "5.Y.Z":
+            return "Windows XP"
+        elif version.minor() == "6.0.Z":
+            return "Windows Vista"
+        elif version.minor() == "6.1.Z":
+            return "Windows 7"
+        elif version.minor() == "6.2.Z":
+            return "Windows 8"
+        elif version.minor() == "6.3.Z":
+            return "Windows 8.1"
+        elif version.minor() == "10.0.Z":
+            return "Windows 10"
+
+    def get_osx_version_name(self, version):
+        if not version:
+            return None
+        elif version.minor() == "10.12.Z":
+            return "Sierra"
+        elif version.minor() == "10.11.Z":
+            return "El Capitan"
+        elif version.minor() == "10.10.Z":
+            return "Yosemite"
+        elif version.minor() == "10.9.Z":
+            return "Mavericks"
+        elif version.minor() == "10.8.Z":
+            return "Mountain Lion"
+        elif version.minor() == "10.7.Z":
+            return "Lion"
+        elif version.minor() == "10.6.Z":
+            return "Snow Leopard"
+        elif version.minor() == "10.5.Z":
+            return "Leopard"
+        elif version.minor() == "10.4.Z":
+            return "Tiger"
+        elif version.minor() == "10.3.Z":
+            return "Panther"
+        elif version.minor() == "10.2.Z":
+            return "Jaguar"
+        elif version.minor() == "10.1.Z":
+            return "Puma"
+        elif version.minor() == "10.0.Z":
+            return "Cheetha"
+
+try:
+    os_info = OSInfo()
+except Exception as exc:
+    logger.error(exc)
+    print("Error detecting os_info")
+
+
+def spt_update_command(sudo=True):
+    """
+        Get the system package tool update command
+    """
+    sudo_str = "sudo " if sudo else ""
```

Comment:
I suggest (it is an open issue, too: https://github.com/conan-io/conan/issues/9), to not hardcode the use of sudo. As it is something that the final consumer defines, the best would be an environment variable, something like CONAN_SYSREQUIRES_SUDO.